test/unit/TPMS-marshal.c

```diff
@@ -317,7 +317,7 @@ tpms_unmarshal_buffer_size_lt_data_nad_lt_offset(void **state)
 {
     TPMS_ALG_PROPERTY alg = {0};
     TPMS_CAPABILITY_DATA cap = {0};
-    uint8_t buffer[sizeof(alg) + sizeof(cap) + 1] = { 0 };
+    uint8_t buffer[sizeof(alg) + sizeof(cap) + 3] = { 0 };
```

Comment:
@diabonas Could you post the gcc-error that leads to this change ?
Looks kind of weird to me ?

Reply:
Sure, the error message is
```
test/unit/TPMS-marshal.c: In function ‘tpms_unmarshal_buffer_size_lt_data_nad_lt_offset’:
test/unit/TPMS-marshal.c:335:9: error: array subscript ‘TPMS_CAPABILITY_DATA[0]’ is partly outside array bounds of ‘uint8_t[1041]’ {aka ‘unsigned char[1041]’} [-Werror=array-bounds]
  335 |     ptr2->capability = HOST_TO_BE_32(TPM2_CAP_ECC_CURVES);
      |         ^~
test/unit/TPMS-marshal.c:320:13: note: while referencing ‘buffer’
  320 |     uint8_t buffer[sizeof(alg) + sizeof(cap) + 1] = { 0 };
      |             ^~~~~~
```
IIUC the problem is that [`ptr2` in l. 334](https://github.com/tpm2-software/tpm2-tss/blob/05beb087939024c2f1b640c58147a0c5e4e94fbf/test/unit/TPMS-marshal.c#L334) is outs of bounds of the `buffer`. I must admit I didn't really do the pointer arithmetic math myself, but `3` is the smallest value that removes the warning message.

Reply:
Seems to make sense though: the first `sizeof(alg)` bytes of the `buffer` are used to store a `TPMS_ALG_PROPERTY` struct (which `ptr` points to), while `ptr2` is a pointer to the offset position `sizeof(alg)+3` of the buffer. Since `ptr2` is supposed to store a `TPMS_CAPABILITY_DATA`, it needs `sizeof(cap)` bytes of space, but `buffer` only offers <code>sizeof(alg)+sizeof(cap)+*1*</code> bytes. In order to have enough room for `ptr2`, we therefore need exactly <code>sizeof(alg)+sizeof(cap)+*3*</code> bytes in `buffer` indeed to avoid a write overflow into adjacent memory.